doc: fix typo on esm loaders example (#47015)
nodejs/node: doc/api/esm.md

    // file, search up the file system for the nearest parent package.json file
    // and read its "type" field.
    const format = await getPackageType(url);

>     // When a hook returns a format of 'commonjs', `source` should be ignored.

Reviewer: ```suggestion
    // When a hook returns a format of 'commonjs', `source` is ignored.
```
